crates/node/src/service/rpc/server.rs

```diff
@@ -77,102 +62,77 @@ pub async fn start_server(
         message_buffer_capacity,
         rpc_api,
         rate_limit,
-        rate_limit_whitelisted_ips,
-        rate_limit_trust_proxy_headers,
     } = config;
 
-    let std_listener = TcpListener::bind(addr)
+    let listener = tokio::net::TcpListener::bind(addr)
         .await
-        .and_then(|a| a.into_std())
-        .with_context(|| format!("binding to address: {addr}"))?;
-    let local_addr = std_listener.local_addr().ok();
-    let host_filter = host_filtering(cors.is_some(), local_addr);
+        .with_context(|| format!("Binding TCP listener to address: {addr}"))?;
+    let local_addr = listener.local_addr().context("Failed to retrieve local address after binding TCP listener")?;
+
+    let ping_config = jsonrpsee::server::PingConfig::new()
+        .ping_interval(Duration::from_secs(30))
+        .inactive_limit(Duration::from_secs(60))
+        .max_failures(3);
 
     let http_middleware = tower::ServiceBuilder::new()
-		.option_layer(host_filter)
-		// Proxy `GET /health` requests to internal `system_health` method.
-		// .layer(ProxyGetRequestLayer::new("/health", "system_health")?)
-        .layer(VersionMiddlewareLayer)
-		.layer(try_into_cors(cors.as_ref())?);
+        .option_layer(host_filtering(cors.is_some(), local_addr))
```

Comment:
There is a potential issue with this logic: if cors is not defined but rpc-external is enabled, the host filtering will reject all external connections. This means that without CORS configured, valid external RPC requests may be unnecessarily blocked.

Reply:
This has been changed so that if `rpc_external` is set, `CORS` will default to be permissive, and otherwise will allow only trafic from localhost. The idea is for it to match how permissive rpc calls are.